pkg/issuercerts/issuercerts.go

```diff
@@ -145,12 +146,21 @@ func CheckCertAlgoRequirements(cert *x509.Certificate) error {
 		if k.Params().BitSize != 256 {
 			return fmt.Errorf("must use P-256 curve for public key, instead P-%d was used", k.Params().BitSize)
 		}
+	} else if cert.IsCA && cert.PublicKeyAlgorithm == x509.RSA {
```

Comment:
I think for the issuer cert `cert.IsCA` is still true, so you'd be allowing here for the issuer cert to also have an RSA public key, which is not the intention, right?

Reply:
You are right! The intention was to do something like `cert.IsRootCA`

I've been trying to find a way (in certs and in go x509 module) to make a difference between a root/intermediate CA and an issuer (CA) cert, but I honestly couldn't.
Initially I thought that the lack of the `Authority Key Identifier` extension could be a good wait to identify a root CA and an intermediate CA, but we have 2 different scenarios here:
- A single cert trust anchor (only the root CA), where we can check that Subject and Issuer names are the same. If they are, it's a root CA cert.
- A multiple cert trust anchor, where one of the certs is a Root one, but there can be 1 (or more) which are as intermediate as the issuer certificate. As we are iterating through all certs in the trust anchor, 1 will be root, but not the rest, and the last cert will be as intermediate CA as the issuer cert...

An easy (but doesn't seem clean) solution would be, from [healthcheck.go](https://github.com/linkerd/linkerd2/blob/main/pkg/healthcheck/healthcheck.go#L997) call a new `CheckIssuerCertAlgoRequirements` which only asserts that the `PublicKeyAlgorithm` is not RSA, and then call the normal `CheckCertAlgoRequirements`. Or (nastier) duplicate the whole `CheckCertAlgoRequirements` with a different logic.

Do you see any other easy way to spot the difference between these types?

Reply:
Hey, apologies for the very late answer!
Indeed, the root cert can itself be (or contain) an intermediate cert, and I don't see a way to tell these two apart. So having the calling sites decide what to check for depending if they're checking the root or the issuer sounds good to me :+1:

Reply:
I think something like this would be it